### DISCUSSION TITLE
What is the best/easiest way to show content base on what's selected in TabStrip control?

### DISCUSSION THREAD

**Discussion by WarWithinMe** [AvaloniaUI/Avalonia#11958] [Q&A] [answered]
I have a VM like this:
```
public class WindowViewModel : ViewModelBase {
    public WindowTabItem[]? Tabs { get; set; }
}
public class WindowTabItem {
    public string? Header { get; set; }
}
```
And I have a Window.axaml like this:
```
<Window
    x:Class="Knight.Views.MainWindow"
    xmlns="https://github.com/avaloniaui"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:vm="using:ViewModels"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    Title="Avalonia.Sample"
    x:CompileBindings="True" x:DataType="vm:WindowViewModel">
    
    <DockPanel>
         <TabStrip ItemsSource="{Binding Tabs}" VerticalAlignment="Center" DockPanel.Dock="Top" />

        <Panel Name="MainViewContainer">
            <!-- When user selects a tab, and some initialization, I want the created content to be shown here. -->
        </Panel>
    </DockPanel>
</Window>
```

For some reason (much effort to customize), I don't want to use a TabControl.
So I use the TabStrip control and bind the VM's `Tabs` to it.
When a user selects a tab, I might need to do some time-consuming
initialization before actually presenting the content.
Can someone please show me some solid code about how to implement this?
Or at least some pseudo code?
I'm thinking this might involve ContentPresenter, but I don't have any experience with it.

Many thanks.

**Comment by timunie** [comment, marked as answer]
place a ContentControl below and work with DataTemplates. https://github.com/AvaloniaUI/Avalonia.Samples#%EF%B8%8F-datatemplate-samples

**Reply by WarWithinMe** [reply]
Thanks for the tip.
